migrations/versions/0435_s_swap_check_for_not_null.py

```diff
@@ -0,0 +1,35 @@
+"""
+
+Revision ID: 0435_s_swap_check_for_not_null
+Revises: 0434_validate_not_null_check
+Create Date: 2023-11-15 22:27:23.511256
+
+"""
+from alembic import op
+from sqlalchemy import column
+
+revision = "0435_s_swap_check_for_not_null"
+down_revision = "0434_validate_not_null_check"
+
+
+def upgrade():
+    # quoth the docs:
+
+    # SET NOT NULL may only be applied to a column provided none of the records in the table contain a NULL value
+    # for the column. Ordinarily this is checked during the ALTER TABLE by scanning the entire table; however, if a
+    # valid CHECK constraint is found which proves no NULL can exist, then the table scan is skipped.
+
+    # so now we can add regular nullable constraints without worrying about a lengthy table scan.
+    op.alter_column("services", "email_sender_local_part", nullable=False)
```

Comment:
there doesn't appear to be a way of guaranteeing that this doesn't hit the lengthy table scan, but we've followed the steps in the postgres docs 🤔